recirq/toric_code/readout_unfolding.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2022 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities to "correct" measured probability distributions using "unfolding."
+
+Based on Nachman et al., "Unfolding quantum computer readout noise"
+npj Quantum Information 6 84 (2020) https://doi.org/10.1038/s41534-020-00309-7
+
+This is similar to using "readout matrix inversion" but more numerically stable.
+Note it does NOT take into account state preparation error, such as stray |1) population
+and pi pulse error.
+
+The readout matrix (R_ij in Nachman et al.) is the *transpose* of the arrangement in
+pyle.dataking.multiple_qubit.readout.correlated_readout_matrix.
+ - Nachman: R_ij = Prob(measured=i | prepared=j) (beginning of "Methods" section)
+    → Columns sum to 1
+ - correlated_readout_matrix: data[row, col] = Prob(measured=col | prepared=row)
+    → Rows sum to 1
+
+We also must be consistent in bitstring ordering / endianness.
```

Comment:
is this a statement of principles? what is the practical consequence

Reply:
I have no idea.  @kjsatz Can you answer?

Reply:
Synced up.  The consistent ordering is needed for compatibility (between differing conventions in pyle and cirq).